src/skmatter/metrics/_pairwise.py

```diff
@@ -205,9 +208,15 @@ def _mahalanobis(
             (cov_inv.shape[0], X.shape[0], Y.shape[0]), order="F"
         )
 
+    _check_dimension(X, cell)
     X, Y = check_pairwise_arrays(X, Y)
     cov_inv = _mahalanobis_preprocess(cov_inv)
     dists = _mahalanobis(cell, X, Y, cov_inv)
     if not squared:
         dists **= 0.5
     return dists
+
+
+def _check_dimension(X, cell):
```

Comment:
Do also want to add a test here that the cell is rectangular? Maybe you have it already and I just oversaw it.

Reply:
Perhaps not, because I do not give a place for the angle in the cell parameter, and every number in the cell will be interpreted as a side length of a rectangular box. I will add a note in the documentation to inform the user that it only supports the rectangular cell.

Reply:
Okay but if the cell is rectangular your format is an 1d array of length 3?

This should be tested and you write in your docs that this is the expected format and if anything else is given your raise a meaningful error message giving the actual type and the type that you expect.

Reply:
Ah sorry, perhaps I should not say rectangular. The cell can be rectangular, cubic, 4-cube or n-cube. Here is not a limitation on the dimension. Thus I only test if the dimension of the cell matches the dimension of descriptors.

Reply:
Yes, rectangular. I also commented this at another place in more details. Maybe it makes sense to rename `cell` to `cell_length` to make sure that what we want.